src/main/java/com/socialmedia/service/UserService.java

```diff
@@ -154,4 +156,24 @@ public void setNewPassword(String forgotPasswordToken, String password) {
     user.getTokensData().setForgotPasswordTokenValidTill(0L);
     jpaRepository.save(user);
   }
+
+  public ApplicationUser deleteFriend(String friendUsername) {
```

Comment:
There are already methods removeFriend and cancelFriendship in UserService, please use them or replace by your implemetation and delete